### PR TITLE
[feature] On-ramp: Add orderProcessor exponential backoff for orders

### PR DESCRIPTION
**Pull request by wachunei**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

This pull request adds exponential backoff functionality to the order polling system. The backoff time will increase with each unknown order status. Additionally, users will still have the ability to force reload the order if necessary.

**Screenshots/Recordings**

~~_If applicable, add screenshots and/or recordings to visualize the before and after of your change_~~

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

### CODE REVIEW

File: app/components/UI/FiatOnRampAggregator/orderProcessor/aggregator.test.ts [hunk 1]
```diff
@@ -312,8 +345,79 @@ describe('processAggregatorOrder', () => {
...
       throw new Error('test error');
     });
+    jest.spyOn(Date, 'now').mockImplementation(() => 1673886669600);
+
     const updatedOrder = await processAggregatorOrder(mockOrder);
     expect(mockGetOrder).toHaveBeenCalledWith('test-id', '0x1234');
     expect(updatedOrder).toEqual(mockOrder);
   });
+
+  it('should return the same object is the error backoff time has not passed', async () => {
```
Comment (Fatxx) [inline app/components/UI/FiatOnRampAggregator/orderProcessor/aggregator.test.ts:355]: Just minor but maybe it's an "if"?

**Review by Fatxx — APPROVED** [review]
LGTM

File: app/components/UI/FiatOnRampAggregator/orderProcessor/aggregator.test.ts (hunk 1 above)
Comment (wachunei) [reply app/components/UI/FiatOnRampAggregator/orderProcessor/aggregator.test.ts:355]: yes! will fix.

Comment (wachunei) [reply app/components/UI/FiatOnRampAggregator/orderProcessor/aggregator.test.ts]: fixed in [e1b2af8](https://github.com/MetaMask/metamask-mobile/pull/6138/commits/e1b2af86afa579eca97f51b95ab38d7b07397f01)

**Comment by bkirb** [comment]
@wachunei I tested this branch and verified there are no regressions. I placed several orders on different chains.

**Pull request merged**